symengine/polys/uintpoly_flint.cpp

```diff
@@ -77,23 +75,15 @@ RCP<const UIntPolyFlint> UIntPolyFlint::from_vec(const RCP<const Symbol> &var,
 
 integer_class UIntPolyFlint::eval(const integer_class &x) const
 {
-    flint::fmpzxx r;
-    fmpz_set_mpz(r._data().inner, get_mpz_t(x));
-    return to_integer_class(static_cast<flint::fmpzxx>(poly_(r)));
+    fz_t r(get_mpz_t(x));
+    return to_integer_class(poly_.eval(r));
 }
 
 vec_integer_class UIntPolyFlint::multieval(const vec_integer_class &v) const
 {
-    flint::fmpz_vecxx t(v.size());
-    for (unsigned int i = 0; i < v.size(); ++i)
-        fmpz_set_mpz(t[i]._data().inner, get_mpz_t(v[i]));
-
-    flint::fmpz_vecxx nn(flint::evaluate(poly_, t));
-
     vec_integer_class res(v.size());
     for (unsigned int i = 0; i < v.size(); ++i)
-        res[i] = to_integer_class(nn[i]);
-
+        res[i] = eval(v[i]);
```

Comment:
[Here](http://www.flintlib.org/flint-2.5.pdf#psge=143) there exists ` fmpz_poly_evaluate_fmpz_vec`, which may have internal optimizations for multi point evaluation. We should use this instead of calling `eval` N times.